ui/src/components/DenylistTable/index.jsx

```diff
@@ -0,0 +1,140 @@
+  static defaultProps = {
+    searchTerm: null,
+  };
+
+  static propTypes = {
+    /** A GraphQL denylisted notifications response. */
+    notificationsConnection: shape({
+      edges: arrayOf(notificationAddress),
+      pageInfo,
+    }).isRequired,
+    onPageChange: func.isRequired,
+    /** A search term to refine the list of notifications */
+    searchTerm: string,
+  };
+
+  state = {
+    sortBy: tableHeaders[1],
+    sortDirection: 'asc',
+  };
+
+  createSortedNotifications = memoize(
+    (notificationsConnection, sortBy, sortDirection) => {
+      // if destination coulumn is clicked sort by notificationAddress property
+      const sortByProperty =
+        camelCase(sortBy) === 'destination'
+          ? 'notificationAddress'
+          : camelCase(sortBy);
+
+      if (!sortBy) {
+        return notificationsConnection;
+      }
+
+      return {
+        ...notificationsConnection,
+        edges: [...notificationsConnection.edges].sort((a, b) => {
+          const firstElement =
+            sortDirection === 'desc'
+              ? b.node[sortByProperty]
+              : a.node[sortByProperty];
+          const secondElement =
+            sortDirection === 'desc'
+              ? a.node[sortByProperty]
+              : b.node[sortByProperty];
+
+          return sort(firstElement, secondElement);
+        }),
+      };
+    },
+    {
+      serializer: ([notificationsConnection, sortBy, sortDirection]) => {
+        const ids = sorted(notificationsConnection.edges);
+
+        return `${ids.join('-')}-${sortBy}-${sortDirection}`;
+      },
+    }
+  );
+
+  handleHeaderClick = sortBy => {
+    const toggled = this.state.sortDirection === 'desc' ? 'asc' : 'desc';
+    const sortDirection = this.state.sortBy === sortBy ? toggled : 'desc';
+
+    this.setState({ sortBy, sortDirection });
+  };
+
+  render() {
+    const { classes, onPageChange, notificationsConnection } = this.props;
+    const { sortBy, sortDirection } = this.state;
+    const sortedNotificationsConnection = this.createSortedNotifications(
+      notificationsConnection,
+      sortBy,
+      sortDirection
+    );
+
+    return (
+      <ConnectionDataTable
+        connection={sortedNotificationsConnection}
+        pageSize={VIEW_DENYLISTED_NOTIFICATIONS_PAGE_SIZE}
+        onHeaderClick={this.handleHeaderClick}
+        onPageChange={onPageChange}
+        headers={tableHeaders}
+        sortByHeader={sortBy}
+        sortDirection={sortDirection}
+        renderRow={({ node }) => (
+          <TableRow key={node.notificationAddress}>
+            <TableCell>
+              <div>
+                <Typography>{node.notificationType}</Typography>
+              </div>
+            </TableCell>
+            <TableCell padding="dense">
+              <Link
+                className={classes.tableCell}
+                to={`/denylist/${encodeURIComponent(
+                  node.notificationAddress
+                )}`}>
```

Comment:
Is it possible for the whole row to be a link?

Reply:
We should aim to be consistent across the UI. For tables with multiple columns, we don't typically have the whole row become a link. I see two options:

1. Have the hover effect on the first column only.
2. Remove the second column and have the whole row be a link. Only do this if the second column is not critical when listing the addresses.

Reply:
I feel we should go with 1 since say, in the future we may have (in an unlike scenario) an irc user and an irc channel with very similar or same names which might cause confusion. Also, sorting would be easier.

@helfi92 @djmitche thoughts? :)

Reply:
Option 1 sounds good to me.

Reply:
Same, and good point about consistency.